src/components/routes: update and refactor RouteItemDisplay and RouteItemEdit components

maceksimon opened the pull request:
Update and refactor `RouteItemDisplay` and `RouteItemEdit` components based on UI design.

__UI design__

* Update layout (elements stacked vertically instead of horizontally)
* Border, border radius and separators
* Use custom icons
* Add empty state to a RouteItemDisplay element (question-mark icon)

__Refactor__

In order to be able to use parts of the route-edit UI in calendar popup panel, refactor these elements into new components: `RouteInputTransportType` and `RouteInputDistance`
* Add tests

tmszi commented on src/components/__tests__/RouteInputDistance.cy.js:
```diff
@@ -0,0 +1,163 @@
+        .should('have.value', i18n.global.t('routes.actionTraceMap'));
+      // input type and input distance are on the same line
+      cy.dataCy('section-input-action').then((element) => {
+        const offsetTop = element[0].offsetTop;
```
These code lines repeat (line 45-52), rewrite as func (or Cypress command), please.

maceksimon replied:
Added as a command.

tmszi commented on src/components/__tests__/RouteInputDistance.cy.js:
```diff
@@ -0,0 +1,163 @@
+      .should('be.visible')
+      .and('have.css', 'font-size', '16px')
+      .and('contain', i18n.global.t('routes.buttonTraceMap'));
+    cy.dataCy('icon-trace-map').should('be.visible');
```
`const iconTraceMap = 'icon-trace-map'`

maceksimon replied:
Extracted to a variable.

tmszi commented on src/components/__tests__/RouteInputDistance.cy.js:
```diff
@@ -0,0 +1,163 @@
+      .and('have.css', 'font-size', '16px')
+      .and('contain', i18n.global.t('routes.buttonTraceMap'));
+    cy.dataCy('icon-trace-map').should('be.visible');
+    cy.dataCy('icon-trace-map').invoke('width').should('be.equal', 24);
```
`const iconTraceMapIconSize = 24`

maceksimon replied:
Extracted to a variable.

tmszi commented on src/components/__tests__/RouteInputTransportType.cy.js:
```diff
@@ -0,0 +1,175 @@
+        .should('be.lessThan', 300);
+      // description and buttons are on the same line
+      cy.dataCy('description-transport').then((element) => {
+        const offsetTop = element[0].offsetTop;
```
Same as my comment above.

maceksimon replied:
Replaced with a test command.

tmszi commented on src/components/__tests__/RouteInputTransportType.cy.js:
```diff
@@ -0,0 +1,175 @@
+      .and('have.color', grey10)
+      .and('contain', i18n.global.t('routes.labelTransportType'));
+    // avatar
+    cy.dataCy('avatar-transport').should('be.visible');
```
`const avatarTransport = 'avatar-transport'`

maceksimon replied:
Extracted to a variable.

tmszi commented on src/components/__tests__/RouteInputTransportType.cy.js:
```diff
@@ -0,0 +1,175 @@
+    // avatar
+    cy.dataCy('avatar-transport').should('be.visible');
+    cy.dataCy('avatar-transport').invoke('height').should('be.eq', 32);
+    cy.dataCy('avatar-transport').invoke('width').should('be.eq', 32);
```
`const avatarTransportIconSize = 32`

maceksimon replied:
Extracted to a variable.

tmszi commented on src/components/__tests__/RouteInputTransportType.cy.js:
```diff
@@ -0,0 +1,175 @@
+    cy.dataCy('avatar-transport').invoke('height').should('be.eq', 32);
+    cy.dataCy('avatar-transport').invoke('width').should('be.eq', 32);
+    // icon
+    cy.dataCy('icon-transport').should('be.visible').and('have.color', primary);
```
Same as my comment above.

maceksimon replied:
Extracted to a variable.

tmszi commented on src/components/__tests__/RouteItemDisplay.cy.js:
```diff
@@ -142,20 +156,124 @@ function coreTests() {
+  });
+
+  it('renders two sections stacked', () => {
+    cy.testElementPercentageWidth(cy.dataCy('section-direction'), 100);
```
```
const sectionDirection = 'section-direction'
const sectionDirectionPercentageWidth = 100
```

maceksimon replied:
Extracted values and selectors into variables.

tmszi commented on src/components/__tests__/RouteItemDisplay.cy.js:
```diff
@@ -142,20 +156,124 @@ function coreTests() {
+
+  it('renders icons correctly', () => {
+    // icon direction
+    cy.dataCy('label-direction-icon', { timeout: 4000 })
```
Really we need here timeout?

maceksimon replied:
This was more of a precaution. Local tests pass without it. Removed.

tmszi commented on src/components/__tests__/RouteInputTransportType.cy.js:
```diff
@@ -0,0 +1,175 @@
+
+  it('renders icons correctly', () => {
+    // icons transport
+    cy.dataCy('icon-transport', { timeout: 4000 }).each((element, index) => {
```
Really we need here timeout?

maceksimon replied:
This was more of a precaution. Local tests pass without it. Removed.

tmszi commented on src/components/__tests__/RouteItemDisplay.cy.js:
```diff
@@ -142,20 +156,124 @@ function coreTests() {
-      .invoke('width')
-      .should('be.eq', 24);
+      .and('have.css', 'font-size', '18px');
+    cy.dataCy('icon-transport').invoke('height').should('be.eq', 18);
```
Repeatable string or number should be const.

maceksimon replied:
Extracted to a variable.

tmszi commented on src/components/__tests__/RouteItemDisplay.cy.js:
```diff
@@ -142,20 +156,124 @@ function coreTests() {
       .invoke('width')
       .should('be.eq', 18);
+    // avatar transport
+    cy.dataCy('avatar-transport').should('be.visible');
```
Repeatable string or number should be const.

maceksimon replied:
Extracted to a variable.

tmszi commented on src/components/__tests__/RouteItemDisplay.cy.js:
```diff
@@ -142,20 +156,124 @@ function coreTests() {
+      // test styles if transport value is logged
+      if (route.transport) {
+        // transport value
+        cy.dataCy('avatar-transport')
```
Repeatable string or number should be const.

maceksimon replied:
Extracted to a variable.

tmszi commented on src/components/__tests__/RouteItemDisplay.cy.js:
```diff
@@ -142,20 +156,124 @@ function coreTests() {
+        retries: 2,
+      });
+    // icon transport
+    cy.dataCy('icon-transport', { timeout: 4000 })
```
Really we need timeout here?

maceksimon replied:
This was more of a precaution. Local tests pass without it. Removed.

tmszi commented on src/components/__tests__/RouteListEdit.cy.js:
```diff
@@ -23,6 +20,10 @@ describe('<RouteListEdit>', () => {
     coreTests();
+
+    it('renders items in two columns', () => {
+      cy.testElementPercentageWidth(cy.dataCy('route-list-item-wrapper'), 50);
```
Repeatable string or number should be const.

maceksimon replied:
Extracted to a variable.

tmszi commented on src/components/routes/RouteInputTransportType.vue:
```diff
@@ -0,0 +1,160 @@
+ * This component is used in `RouteItemEdit` and `RouteCalendarPanel`.
+ *
+ * @props
+ * - `modelValue` (string, required): The object representing selected value.
```
Missing prop doc.

maceksimon replied:
Added.

tmszi commented on src/i18n/sk.toml:
```diff
@@ -561,3 +561,4 @@ car = "Autom"
 none = "Nič"
```
Žiadna

maceksimon replied:
Corrected.

tmszi commented:
Custom icons path should be ./public/icons/routes_calendar/icons.svg

maceksimon commented:
SVG icons are moved to the correct folder.

tmszi commented on src/components/__tests__/RouteItemDisplay.cy.js:
```diff
@@ -15,147 +43,272 @@ describe('<RouteItemDisplay>', () => {
+    cy.dataCy(selectorDirectionIcon)
+      .should('be.visible')
+      .matchImageSnapshot(`${Cypress.currentTest.titlePath}-direction`, {
+        failureThreshold: 0.1,
```
params args repeat across tests

maceksimon replied:
Extracted function parameter to a variable.

tmszi commented:
Pointer with "do not select" icon, is allowing select route in the calendar.

![route_selection_on_deactivated_route](https://github.com/user-attachments/assets/9ee2d0b4-375f-41c9-b7d8-3f2536e63c9e)

maceksimon commented:
> Pointer with "do not select" icon, is allowing select route in the calendar.

Fixed. Items outside current month are disabled via a prop in `CalendarItemDisplay`.

maceksimon commented:
> It is possible typing negative number or zero into Input distance input widget (By bike, On foot, By public transport). Should not be here a input value validator?

Updated. Added a conditional validation (enabled by default). This will most likely be used in [list section](https://www.figma.com/design/L8dVREySVXxh3X12TcFDdR/Do-pr%C3%A1ce-na-kole?node-id=4858-104042&t=30orbYgabbcBdpGQ-1) where user can edit missing routes (we might want to allow logging only some of them).

tmszi commented:
> > It is possible typing negative number or zero into Input distance input widget (By bike, On foot, By public transport). Should not be here a input value validator?
> 
> Updated. Added a conditional validation (enabled by default). This will most likely be used in [list section](https://www.figma.com/design/L8dVREySVXxh3X12TcFDdR/Do-pr%C3%A1ce-na-kole?node-id=4858-104042&t=30orbYgabbcBdpGQ-1) where user can edit missing routes (we might want to allow logging only some of them).

I tried again, but input widget allow me typing negative number. Is it really working as expected?

maceksimon commented:
> I tried again, but input widget allow me typing negative number. Is it really working as expected?

My mistake, negative numbers were not validated. Added validation + test.
